### ISSUE TITLE
Potential flaky tests in the project

### ISSUE THREAD

**Issue by ShangwenWang** [google/guava#5865]
Dear developers,

Recently I used this project to do something. But when I built it 

- com.google.common.util.concurrent.UninterruptibleMonitorTest#testEnterWhen_initiallyFalse
- com.google.common.cache.PopulatedCachesTest#testKeySet_populated

It was surprising that when I rebuilt it, they are both passed. So I guess they may be flaky tests. 
I report this phenomenon to you and attached is the test report I obtained. I would appreciate it if you could pay attention to them.

[test_reports.txt](https://github.com/google/guava/files/7843033/test_reports.txt)

**Comment by cgdecker** [comment]
Yes, it does look like these are tests that could be flaky. I think it's unlikely that we're going try to do anything about that, though.

**Comment by ShangwenWang** [comment]
Thanks for your reply. It would be nice if someone can look into it in the future.
